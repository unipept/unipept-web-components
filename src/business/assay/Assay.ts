import { CountTable } from "@/business/counts/CountTable";
import Visitable from "@/logic/patterns/visitor/Visitable";

export default abstract class Assay implements Visitable<AssayVisitor> {
    protected constructor(
        public readonly changeListeners: ChangeListener<Assay>[],
        public readonly id: string,
        protected name?: string,
        protected date?: Date,
    ) {}

    public getId(): string {
        return this.id;
    }

    public getName() {
        return this.name;
    }

    public setName(name: string) {
        const oldName: string = this.name;
        this.name = name;
        this.onUpdate("name", oldName, name);
    }

    public getDate() {
        return this.date;
    }

    public setDate(date: Date) {
        const oldDate: Date = date;
        this.date = date;
        this.onUpdate("date", oldDate, date);
    }

    protected onUpdate(field: string, oldValue: any, newValue: any) {
        for (const changeListener of this.changeListeners) {
            changeListener.onChange(this, field, oldValue, newValue);
        }
    }

    abstract async accept(visitor: AssayVisitor): Promise<void>;
}

