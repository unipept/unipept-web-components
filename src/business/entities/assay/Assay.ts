import ChangeListener from "./../ChangeListener";
import AssayVisitor from "./AssayVisitor";
import Visitable from "./../../visitor/Visitable";

export default abstract class Assay implements Visitable<AssayVisitor> {
    private name: string = "";
    private date: Date = new Date();
    private changeListeners: ChangeListener<Assay>[] = [];

    protected constructor(
        public id: string
    ) {}

    public addChangeListener(listener: ChangeListener<Assay>): void {
        this.changeListeners.push(listener);
    }

    public getId(): string {
        return this.id;
    }

    public getName() {
        return this.name;
    }

    public setName(name: string) {
        const oldName: string = this.name;
        this.name = name;
        if (oldName !== name) {
            this.onUpdate("name", oldName, name);
        }
    }

    public getDate() {
        return this.date;
    }

    public setDate(date: Date) {
        const oldDate: Date = date;
        this.date = date;
        if (!oldDate || date.getTime() !== oldDate.getTime()) {
            this.onUpdate("date", oldDate, date);
        }
    }

    protected onUpdate(field: string, oldValue: any, newValue: any) {
        for (const changeListener of this.changeListeners) {
            changeListener.onChange(this, field, oldValue, newValue);
        }
    }

    abstract async accept(visitor: AssayVisitor): Promise<void>;
}

