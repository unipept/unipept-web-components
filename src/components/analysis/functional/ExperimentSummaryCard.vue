<template>
    <v-card style="min-height: 100%; display: flex; flex-direction: column;">
        <card-header>
            <card-title>
                Experiment Summary
            </card-title>
        </card-header>
        <v-card-text style="flex-grow: 1; display: flex; flex-direction: column;">
            <search-settings-form
                :disabled="$store.getters.selectedDatasets.some(el => el.progress !== 1)"
                :equate-il.sync="equateIl"
                :filter-duplicates.sync="filterDuplicates"
                :missing-cleavage.sync="missingCleavage"
                style="flex-grow: 1;">
            </search-settings-form>
            <div class="card-actions" >
                <tooltip message="Restart search with selected samples using the settings chosen above.">
                    <v-btn :disabled="$store.getters.selectedDatasets.some(el => el.progress !== 1)" @click="reprocess()" color="primary"><v-icon left>mdi-restore</v-icon>Update</v-btn>
                </tooltip>
            </div>
            <v-divider></v-divider>
            <span v-if="!$store.getters.activeDataset">No dataset is selected... Wait for at least one dataset to be loaded or select one.</span>
            <span v-else>
                We managed to match {{ $store.getters.matchedPeptides }} of your {{ $store.getters.searchedPeptides }} peptides.
                Unfortunately, <a style="cursor: pointer;" @click="showNotFoundPeptidesModal">{{ $store.getters.missedPeptides.length }}</a> peptides couldn't be found.
            </span>
        </v-card-text>
        <v-dialog v-model="showNotFoundModal" :width="600">
            <v-card>
                <v-card-title>
                    {{ $store.getters.missedPeptides.length }} missed peptides
                </v-card-title>
                <v-card-text>
                    <v-row>
                        <v-col :cols="7">
                            <span>
                                Sorry, we didn't manage to find {{ $store.getters.missedPeptides.length }} of your 
                                peptides. You can BLAST them by clicking the links or copy them by using the button on 
                                the right.
                            </span>
                        </v-col>
                        <v-col :cols="5">
                            <v-btn @click="copyToClipboard">
                                <v-icon left>
                                    mdi-clipboard-text-outline
                                </v-icon>
                                Copy to clipboard
                            </v-btn>
                        </v-col>
                    </v-row>
                    
                    <ul>
                        <li v-for="missed of $store.getters.missedPeptides" :key="missed">
                            <a :href="getBlastUrl(missed)" target="_blank">
                                {{ missed }}
                            </a>
                        </li>
                    </ul>
                </v-card-text>
            </v-card>
        </v-dialog>
    </v-card>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop, Watch } from "vue-property-decorator";
import SearchSettingsForm from "../SearchSettingsForm.vue";

import CardHeader from "../../custom/CardHeader.vue";
import CardTitle from "../../custom/CardTitle.vue";

import Clipboard from "clipboard";
import TaxaDataSource from "../../../logic/data-source/TaxaDataSource";
import PeptideContainer from "../../../logic/data-management/PeptideContainer";
import Tooltip from "../../custom/Tooltip.vue";
import * as clipboard from "clipboard-polyfill";

@Component({
    components: { CardTitle, CardHeader, SearchSettingsForm, Tooltip }
})
export default class ExperimentSummaryCard extends Vue {
    private equateIl: boolean = true;
    private filterDuplicates: boolean = true;
    private missingCleavage: boolean = false;
    private showNotFoundModal: boolean = false;

    created() {
        this.equateIl = this.$store.getters.searchSettings.il;
        this.filterDuplicates = this.$store.getters.searchSettings.dupes;
        this.missingCleavage = this.$store.getters.searchSettings.missed;
    }

    reprocess(): void {
        this.$store.dispatch("setSearchSettings", { il: this.equateIl, dupes: this.filterDuplicates, missed: this.missingCleavage });

        this.$store.dispatch("setActiveDataset", null);
        let promises: Promise<any>[] = [];
        for (let dataset of this.$store.getters.selectedDatasets) {
            promises.push(this.$store.dispatch("processDataset", dataset));
        }
    }

    private copyToClipboard() {
        clipboard.writeText(this.$store.getters.missedPeptides.join("\n"));
    }

    private showNotFoundPeptidesModal() {
        this.showNotFoundModal = true;
    }

    private getBlastUrl(peptide: string) {
        return "http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch&SET_SAVED_SEARCH=on" +
            "&USER_FORMAT_DEFAULTS=on&PAGE=Proteins&PROGRAM=blastp&QUERY=" + peptide + "&GAPCOSTS=11%201" + 
            "&EQ_MENU=Enter%20organism%20name%20or%20id--completions%20will%20be%20suggested&DATABASE=nr" +
            "&BLAST_PROGRAMS=blastp&MAX_NUM_SEQ=100&SHORT_QUERY_ADJUST=on&EXPECT=10&WORD_SIZE=3" + 
            "&MATRIX_NAME=BLOSUM62&COMPOSITION_BASED_STATISTICS=2&SHOW_OVERVIEW=on&SHOW_LINKOUT=on" + 
            "&ALIGNMENT_VIEW=Pairwise&MASK_CHAR=2&MASK_COLOR=1&GET_SEQUENCE=on&NEW_VIEW=on&NUM_OVERVIEW=100" + 
            "&DESCRIPTIONS=100&ALIGNMENTS=100&FORMAT_OBJECT=Alignment&FORMAT_TYPE=HTML&OLD_BLAST=false"
    }
}
</script>

<style lang="less">
    @import './../../../assets/style/card.css.less';
</style>
