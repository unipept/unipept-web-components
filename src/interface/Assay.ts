export default interface Assay {
    id: string
    name: string
    peptides: string[]
    amountOfPeptides: number
    createdAt: Date
};
