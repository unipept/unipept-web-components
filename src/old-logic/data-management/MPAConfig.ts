export default interface MPAConfig {
  il: boolean;
  dupes: boolean;
  missed: boolean;
}
