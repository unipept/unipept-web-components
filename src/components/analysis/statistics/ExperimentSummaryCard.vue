<docs>
The ExperimentSummaryCard summarizes an experiment into statistics: mainly the amount of peptides that were found, the 
peptides that were not found (including the ability to show these as a list). This component also allows the user to
change the currently active search settings and redo the analysis of all selected assays.
</docs>

<template>
    <v-card style="min-height: 100%; display: flex; flex-direction: column;">
        <card-header>
            <card-title>
                Experiment Summary
            </card-title>
        </card-header>
        <v-card-text style="flex-grow: 1; display: flex; flex-direction: column;">
            <search-settings-form
                :disabled="$store.getters.selectedDatasets.some(el => el.progress !== 1)"
                :equate-il.sync="equateIl"
                :filter-duplicates.sync="filterDuplicates"
                :missing-cleavage.sync="missingCleavage"
                style="flex-grow: 1;">
            </search-settings-form>
            <div class="card-actions" >
                <tooltip message="Restart search with selected samples using the settings chosen above.">
                    <v-btn :disabled="$store.getters.selectedDatasets.some(el => el.progress !== 1)" @click="reprocess()" color="primary">
                        <v-icon left>
                            mdi-restore
                        </v-icon>
                        Update
                    </v-btn>
                </tooltip>
                <tooltip message="Download a CSV-file with the results of this analysis.">
                    <v-btn :disabled="$store.getters.selectedDatasets.some(el => el.progres !== 1)" @click="downloadCsv()" color="default">
                        <v-icon>
                            mdi-download
                        </v-icon>
                        Download results
                    </v-btn>
                </tooltip>
            </div>
            <v-divider></v-divider>
            <span v-if="!activeAssay">No dataset is selected... Wait for at least one dataset to be loaded or select one.</span>
            <span v-else>
                We managed to match {{ matchedPeptides }} of your {{ searchedPeptides }} peptides.
                Unfortunately, <a style="cursor: pointer;" @click="showNotFoundPeptidesModal">{{ missedPeptides.length }}</a> peptides couldn't be found.
            </span>
        </v-card-text>
        <v-dialog v-model="showNotFoundModal" :width="600">
            <v-card>
                <v-card-title>
                    {{ missedPeptides.length }} missed peptides
                </v-card-title>
                <v-card-text>
                    <missing-peptides-list :dataset="activeAssay">
                    </missing-peptides-list>
                </v-card-text>
            </v-card>
        </v-dialog>
    </v-card>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop, Watch } from "vue-property-decorator";
import SearchSettingsForm from "../SearchSettingsForm.vue";

import CardHeader from "../../custom/CardHeader.vue";
import CardTitle from "../../custom/CardTitle.vue";
import MissingPeptidesList from "./MissingPeptidesList.vue";

import Clipboard from "clipboard";
import TaxaDataSource from "../../../logic/data-source/TaxaDataSource";
import PeptideContainer from "../../../logic/data-management/PeptideContainer";
import Tooltip from "../../custom/Tooltip.vue";
import Assay from "../../../logic/data-management/assay/Assay";

@Component({
    components: { CardTitle, CardHeader, SearchSettingsForm, Tooltip, MissingPeptidesList }
})
export default class ExperimentSummaryCard extends Vue {
    /**
     * Is the user able to interact with this component? (True if he can interact, false otherwise).
     */
    @Prop({ required: false, default: false })
    private disabled: boolean;
    /**
     * Denotes the assay that's currently been selected by the user (this is the assay for which the visualizations
     * are visible at this moment).
     */
    @Prop({ required: true })
    private activeAssay: Assay;

    private equateIl: boolean = true;
    private filterDuplicates: boolean = true;
    private missingCleavage: boolean = false;
    private showNotFoundModal: boolean = false;

    private searchedPeptides: number = 0;
    private matchedPeptides: number = 0;
    private missedPeptides: string[] = [];

    private loading: boolean = true;

    created() {
        this.equateIl = this.$store.getters.searchSettings.il;
        this.filterDuplicates = this.$store.getters.searchSettings.dupes;
        this.missingCleavage = this.$store.getters.searchSettings.missed;
    }

    reprocess(): void {
        /**
         * Fired after the user chose new settings and decided to rerun the analysis on all selected assays.
         */
        this.$emit("update-search-settings", { il: this.equateIl, dupes: this.filterDuplicates, missed: this.missingCleavage });
    }

    private showNotFoundPeptidesModal() {
        this.showNotFoundModal = true;
    }

    @Watch("activeAssay")
    private async onActiveDatasetChanged(): Promise<void> {
        if (this.activeAssay) {
            this.loading = true;
            let taxaSource: TaxaDataSource = await this.activeAssay.dataRepository.createTaxaDataSource();
            this.searchedPeptides = await taxaSource.getAmountOfSearchedPeptides();
            this.matchedPeptides = await taxaSource.getAmountOfMatchedPeptides();
            this.missedPeptides = await taxaSource.getMissedPeptides();
            this.loading = false;
        }
    }

    private async downloadCsv(): Promise<void> {
        if (this.activeAssay) {
            const taxaDataSource: TaxaDataSource = await this.activeAssay.dataRepository.createTaxaDataSource();
            await taxaDataSource.toCSV();
        }
    }
}
</script>

<style lang="less">
    @import './../../../assets/style/card.css.less';
</style>
