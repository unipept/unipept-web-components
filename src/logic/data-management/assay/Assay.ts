import DataRepository from "../../data-source/DataRepository";
import Visitable from "../../patterns/visitor/Visitable";
import AssayVisitor from "src/logic/data-management/assay/AssayVisitor";
import { StorageType } from "../StorageType";
import MPAConfig from "../MPAConfig";
import Entity from "./Entity";
import ChangeListener from "@/logic/data-management/ChangeListener";

export default abstract class Assay implements Visitable<AssayVisitor>, Entity<string> {
    protected id: string;
    protected name: string;
    protected date: Date;
    protected changeListener: ChangeListener;
    protected storageType: StorageType;

    protected _dataRepository: DataRepository;

    public progress: number = 0;

    constructor(changeListener?: ChangeListener, id?: string, storageType?: StorageType, name?: string, date?: Date) {
        this.id = id;
        this.name = name;
        this.date = date;
        this.changeListener = changeListener;
        this.storageType = storageType;
    }

    getId(): string {
        return this.id;
    }

    setId(id: string) {
        this.id = id;
        this.changeListener.onChange("id");
    }

    getName() {
        return this.name;
    }

    setName(name: string) {
        this.name = name;
        this.changeListener.onChange("name");
    }

    getDate() {
        return this.date;
    }

    setDate(date: Date) {
        this.date = date;
        this.changeListener.onChange("date");
    }

    getStorageType() {
        return this.storageType;
    }

    setStorageType(storageType: StorageType) {
        this.storageType = storageType;
    }

    get dataRepository() {
        return this._dataRepository;
    }

    getDateFormatted(): string {
        // @ts-ignore
        return this.date.getFullYear() + "/" + (this.date.getMonth() + 1).toString().padStart(2, "0") + "/" + this.date.getDate().toString().padStart(2, "0");
    }

    abstract async initDataRepository(mpaConfig: MPAConfig, baseUrl: string);
    abstract async accept(visitor: AssayVisitor): Promise<void>;
}
