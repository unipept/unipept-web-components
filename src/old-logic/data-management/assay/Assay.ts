import DataRepository from "../../data-source/DataRepository";
import Visitable from "../../patterns/visitor/Visitable";
import AssayVisitor from "./AssayVisitor";
import { StorageType } from "../StorageType";
import MPAConfig from "../MPAConfig";
import Entity from "./Entity";
import ChangeListener from "./../ChangeListener";

export default abstract class Assay implements Visitable<AssayVisitor>, Entity<string> {
    protected readonly id: string;
    protected name: string;
    protected date: Date;
    protected changeListeners: ChangeListener<Assay>[];
    protected storageType: StorageType;

    protected _dataRepository: DataRepository;

    public progress: number = 0;

    constructor(changeListeners: ChangeListener<Assay>[], id: string, storageType?: StorageType, name?: string, date?: Date) {
        this.id = id;
        this.name = name;
        this.date = date;
        this.changeListeners = changeListeners;
        this.storageType = storageType;
    }

    getId(): string {
        return this.id;
    }

    getName() {
        return this.name;
    }

    setName(name: string) {
        const oldName: string = this.name;
        this.name = name;
        this.onUpdate("name", oldName, name);
    }

    getDate() {
        return this.date;
    }

    setDate(date: Date) {
        const oldDate: Date = date;
        this.date = date;
        this.onUpdate("date", oldDate, date);
    }

    getStorageType() {
        return this.storageType;
    }

    setStorageType(storageType: StorageType) {
        this.storageType = storageType;
    }

    get dataRepository() {
        return this._dataRepository;
    }

    getDateFormatted(): string {
        // @ts-ignore
        return this.date.getFullYear() + "/" + (this.date.getMonth() + 1).toString().padStart(2, "0") + "/" + this.date.getDate().toString().padStart(2, "0");
    }

    public reset(): void {
        this.progress = 0;
        this._dataRepository = undefined;
    }

    protected onUpdate(field: string, oldValue: any, newValue: any) {
        for (const changeListener of this.changeListeners) {
            changeListener.onChange(this, field, oldValue, newValue);
        }
    }

    abstract async initDataRepository(mpaConfig: MPAConfig, baseUrl: string);
    abstract async accept(visitor: AssayVisitor): Promise<void>;
}
