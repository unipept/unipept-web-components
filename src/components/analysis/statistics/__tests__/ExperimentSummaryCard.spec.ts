import { shallowMount, createLocalVue, mount, Wrapper } from "@vue/test-utils";
import ExperimentSummaryCard from "./../ExperimentSummaryCard.vue";
import Vue from "vue";
import Vuetify from "vuetify";
import Vuex, { Store } from "vuex";
import Mock from "@/test/Mock";
import Assay from "@/logic/data-management/assay/Assay";
import flushPromises from "flush-promises";
import TestUtils from "@/test/TestUtils";
import { CombinedVueInstance } from "vue/types/vue";
import MPAConfig from "@/logic/data-management/MPAConfig";

Vue.use(Vuetify);
Vue.use(Vuex);

const localVue = createLocalVue();

// These peptides should be found by the tests
const foundPeptides: string[] = [
    "SGLAPFYSDK",
    "AAANESFGYNEDELVSSDLVGMR",
    "LAEEVLR",
    "YEVGTMLELPR",
    "TNTLLQSAFFK",
    "ANFEGECSEVGMYLAMAR",
    "MEVAVGDK",
    "LLDLGVLVGSGYHVNPK",
    "VYFLNFKPESSDEWK",
    "LGLVAVSR"
];

describe("ExperimentSummaryCard", () => {
    let vuetify;
    let getters;
    let store;

    beforeEach(() => {
        vuetify = new Vuetify();
    });

    it("correctly renders all items that are currently present", async(done) => {
        const wrapper = await mountWithValidDatasets(vuetify);

        await TestUtils.sleep(1000);
        await flushPromises();

        expect(wrapper.find(".peptide-match-text").html()).toMatchSnapshot();
        done();
    });

    it("correctly renders a placeholder when no dataset is selected", async(done) => {
        const wrapper = await mountWithoutDatasets(vuetify);
        expect(wrapper.find(".dataset-placeholder-text").html()).toMatchSnapshot();
        done();
    });

    it("correctly updates search settings when clicking update", async(done) => {
        const mock: Mock = new Mock();
        const assay: Assay = await mock.mockInitializedAssay();

        const actions = {
            setSearchSettings: jest.fn(),
            setActiveDataset: jest.fn(),
            processDataset: jest.fn(),
        }

        const getters = {
            selectedTerm: () => "Organism",
            searchSettings: () => {
                return {
                    il: true,
                    dupes: true,
                    missed: false
                }
            },
            selectedDatasets: () => [assay],
            activeDataset: () => assay,
        };

        const store = new Vuex.Store({
            actions,
            getters
        });

        const wrapper = await mount(ExperimentSummaryCard, {
            store,
            localVue,
            vuetify
        });

        wrapper.find(".card-actions button").trigger("click");
        // Check that the function did inform the store of the current values
        expect(actions.setSearchSettings).toBeCalled();
        expect(actions.setSearchSettings.mock.calls[0][1]).toEqual({
            il: true,
            dupes: true,
            missed: false
        });

        actions.setSearchSettings.mockReset();

        // Now we change the search settings-values
        wrapper.find(".search-settings-form input").trigger("click");
        wrapper.find(".card-actions button").trigger("click");

        // Check that different search settings are now being used in the store.
        expect(actions.setSearchSettings).toBeCalled();
        expect(actions.setSearchSettings.mock.calls[0][1]).toEqual({
            il: false,
            dupes: true,
            missed: false
        });

        done();
    });

    it("opens a dialog when clicking the missed peptides text", async(done) => {
        // Fix for Vuetify warn in combination with Jest and Dialogs
        const app = document.createElement("div");
        app.setAttribute("data-app", "");
        document.body.append(app);

        const wrapper = await mountWithValidDatasets(vuetify);
    
        expect(wrapper.find(".v-dialog--active").exists()).toBeFalsy();
        wrapper.find(".peptide-match-text a").trigger("click");
        expect(wrapper.find(".v-dialog--active").exists()).toBeTruthy();

        done();
    });
});

const getValidStore = async function(): Promise<Store<unknown>> {
    const mock: Mock = new Mock();
    const assay: Assay = await mock.mockInitializedAssay();
    const getters = {
        selectedTerm: () => "Organism",
        searchSettings: () => {
            return {
                il: true,
                dupes: true,
                missed: false
            }
        },
        selectedDatasets: () => [assay],
        activeDataset: () => assay,
    };

    const store = new Vuex.Store({
        getters
    });

    return store;
}

/**
 * Mount the ExperimentSummaryCard-component with a valid lists of assays in the store and a valid activeDataset
 * entry in the Vuex store.
 * 
 * @param vuetify A valid Vuetify instance that should be used during the tests.
 */
const mountWithValidDatasets = async function(vuetify): Promise<Wrapper<CombinedVueInstance<ExperimentSummaryCard, object, object, object, Record<never, any>>>> {
    const store = await getValidStore();

    return mount(ExperimentSummaryCard, {
        store,
        localVue,
        vuetify
    });
}

/**
 * Mount the ExperimentSummaryCard-component without valid assay.
 * 
 * @param vuetify A valid Vuetify instance that should be used during the tests.
 */
const mountWithoutDatasets = async function(vuetify): Promise<Wrapper<CombinedVueInstance<ExperimentSummaryCard, object, object, object, Record<never, any>>>> {
    const getters = {
        selectedTerm: () => "Organism",
        searchSettings: () => {
            return {
                il: true,
                dupes: true,
                missed: false
            }
        },
        selectedDatasets: () => [],
        activeDataset: () => null,
    };

    const store = new Vuex.Store({
        getters
    });

    return mount(ExperimentSummaryCard, {
        store,
        localVue,
        vuetify
    });
}
