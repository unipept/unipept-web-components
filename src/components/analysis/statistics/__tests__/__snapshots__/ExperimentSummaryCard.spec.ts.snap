// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentSummaryCard correctly renders a placeholder when no dataset is selected 1`] = `
"<span class=\\"dataset-placeholder-text\\">
            No dataset is selected... Wait for at least one dataset to be loaded or select one.
        </span>"
`;

exports[`ExperimentSummaryCard correctly renders all items that are currently present 1`] = `
"<span class=\\"peptide-match-text\\">
            We managed to match 15 of your 25
            peptides. Unfortunately,
            <a style=\\"cursor: pointer;\\">
                10
            </a>
            peptides couldn't be found.
        </span>"
`;

exports[`ExperimentSummaryCard opens a dialog when clicking the missed peptides text 1`] = `
"<span class=\\"peptide-match-text\\">
            We managed to match 15 of your 25
            peptides. Unfortunately,
            <a style=\\"cursor: pointer;\\">
                10
            </a>
            peptides couldn't be found.
        </span>"
`;
